### PR TITLE
Fix grouping example

### PR DESCRIPTION
**Pull request by amanmahajan7**
- [x] Fix grouping example
- [x] Fix cell actions example https://github.com/adazzle/react-data-grid/issues/1342 
- [x] Allow draggable columns to be frozen 

image

### CODE REVIEW

**Comment by malonecj** [comment]
Hey @amanmahajan7 Since this is a minor change we should have master as the base branch rather than next

**Comment by amanmahajan7** [comment]
@malonecj master does not have the latest code, we should catchup with next and I can create a new PR with master as the base branch

File: packages/react-data-grid-addons/src/data/RowGrouper.js [hunk 1]
```diff
@@ -22,14 +22,15 @@ class RowGrouper {
...
+    const columnGroupDisplayName = this.columns.length > 0 && typeof this.columns[columnIndex] === 'string' ? this.columns[columnIndex] : this.columns[columnIndex].name;
     let groupedRows = this.resolver.getGroupedRows(rows, columnName);
     let keys = this.resolver.getGroupKeys(groupedRows);
     let dataviewRows = this.resolver.initRowsCollection();
 
     for (let i = 0; i < keys.length; i++) {
       let key = keys[i];
       let isExpanded = this.isRowExpanded(columnName, key);
-      let rowGroupHeader = {name: key, __metaData: {isGroup: true, treeDepth: columnIndex, isExpanded: isExpanded, columnGroupName: columnName}};
+      let rowGroupHeader = {name: key, __metaData: {isGroup: true, treeDepth: columnIndex, isExpanded: isExpanded, columnGroupName: columnName, columnGroupDisplayName}};
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/data/RowGrouper.js]: Added this to show `column.name` instead if `column.key` in the group row

File: packages/react-data-grid-addons/src/draggable/DraggableHeaderCell.js [hunk 2]
```diff
@@ -1,40 +1,30 @@
...
-    let img = new Image();
+    const { connectDragPreview } = this.props;
+    const img = new Image();
     img.src = './assets/images/drag_column_full.png';
-    img.onload = function() {
-      connectDragPreview(img);
-    };
-  }
-
-  setScrollLeft(scrollLeft) {
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/draggable/DraggableHeaderCell.js]: This is no longer required as `HeaderCell` content is wrapped inside a `DraggableHeaderCell`

File: packages/react-data-grid-examples/src/scripts/example12-customRowRenderer.js [hunk 3]
```diff
@@ -10,11 +10,6 @@ class RowRenderer extends React.Component {
     renderBaseRow: PropTypes.func.isRequired
   };
 
-  setScrollLeft = (scrollBy) => {
```
Comment (amanmahajan7) [inline packages/react-data-grid-examples/src/scripts/example12-customRowRenderer.js]: This is not required as we introduced `renderBaseRow`

File: packages/react-data-grid/src/Canvas.js [hunk 4]
```diff
@@ -1,443 +1,441 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-import PropTypes from 'prop-types';
-import Row from './Row';
-import cellMetaDataShape from 'common/prop-shapes/CellActionShape';
-import * as rowUtils from './RowUtils';
-import RowGroup, { DefaultRowGroupRenderer } from './RowGroup';
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/Canvas.js]: `DefaultRowGroupRenderer ` does not exist

File: packages/react-data-grid/src/Canvas.js [hunk 5]
```diff
@@ -1,443 +1,441 @@
...
+    if (row) {
+      const node = ReactDOM.findDOMNode(row);
+      return node && node.clientHeight > 0 ? node.clientHeight : this.props.rowHeight;
+    }
+    return this.props.rowHeight;
+  }
+
+  getSelectedRowColumns = (rowIdx) => {
+    const row = this.getRowByRef(rowIdx);
+    return row && row.props ? row.props.columns : this.props.columns;
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/Canvas.js]: This was causing an error

File: packages/react-data-grid/src/HeaderCell.js [hunk 6]
```diff
@@ -83,42 +87,54 @@ class HeaderCell extends React.Component {
...
+    }, this.props.className, column.cellClass);
+    const cell = (
       <div className={className} style={this.getStyle()}>
-        {cell}
+        {this.getCell()}
         {resizeHandle}
       </div>
     );
+
+    if (rowType === HeaderRowType.HEADER && column.draggable) {
```
Comment (amanmahajan7) [inline packages/react-data-grid/src/HeaderCell.js]: Wrapping `HeaderCell` content. This has a few benefits
1) `setScrollLeft` is not required in `DraggableHeaderCell`. So this works with frozen columns, it was broken before
2) It works with any header cell renderer

In the future I would like to clean up header rendering logic so we can easily compose them

```
DraggableHeaderCell(SortableHeaderCell(SimpleHeaderCell or column.headerRenderer))
```

File: packages/react-data-grid-addons/src/draggable/DraggableHeaderCell.js [hunk 7]
```diff
@@ -1,40 +1,30 @@
...
+    img.onload = () => connectDragPreview(img);
   }
 
   render() {
-    const { connectDragSource, isDragging} = this.props;
+    const { connectDragSource, isDragging } = this.props;
     if (isDragging) {
       return null;
     }
-    return connectDragSource(<div ref={node => this.node = node} style={{cursor: 'move'}}>{this.props.renderHeaderCell(this.props)}</div>);
```
Comment (amanmahajan7) [inline packages/react-data-grid-addons/src/draggable/DraggableHeaderCell.js]: `renderHeaderCell` is not passed as a prop

**Pull request merged**